### PR TITLE
Feature/support embeds

### PR DESCRIPTION
**Pull request by tdakanalis**
Features:
- Adds support for migrating the inline assets/files discovered within GitHub issues and comments as Radicle embeds.

### CODE REVIEW

File: src/main/java/network/radicle/tools/github/utils/FileUtils.java [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.util.Arrays;
+import java.util.Base64;
+
+public class FileUtils {
```
Comment (JChrist) [inline src/main/java/network/radicle/tools/github/utils/FileUtils.java:16]: Ideally this should be refactored to a new service

File: src/main/java/network/radicle/tools/github/utils/Markdown.java [hunk 2]
```diff
@@ -61,8 +80,52 @@ public static String escape(String markdown) {
         return Strings.nullToEmpty(markdown).replace(":", "\\:");
     }
 
+    public static List<MarkdownLink> extractUrls(String markdown) {
```
Comment (JChrist) [inline src/main/java/network/radicle/tools/github/utils/Markdown.java:83]: Ideally this class would get refactored to a markdown service

**Review by JChrist — CHANGES_REQUESTED** [review]

File: src/main/java/network/radicle/tools/github/clients/GitHubClient.java [hunk 3]
```diff
@@ -115,4 +121,48 @@ public Commit getCommit(String commitId) throws Exception {
...
+            logger.warn("Failed to fetch: {}", url);
+            return null;
+        }
+    }
+
+    private byte[] followRedirects(String url, boolean withAuthorization) throws IOException {
+        try (var response = client.target(url)
+                .request()
+                .header(HttpHeaders.AUTHORIZATION, withAuthorization ? "Bearer " + Strings.nullToEmpty(config.getGithub().token()) : null)
+                .cookie("user_session", config.getGithub().session())
```
[the remark this answers is not in the document]
Comment (tdakanalis) [reply src/main/java/network/radicle/tools/github/clients/GitHubClient.java]: @JChrist Fixed. Could you review the changes please?

File: src/main/java/network/radicle/tools/github/utils/FileUtils.java (hunk 1 above)
Comment (tdakanalis) [reply src/main/java/network/radicle/tools/github/utils/FileUtils.java]: @JChrist Fixed. Could you review the changes please?

File: src/main/java/network/radicle/tools/github/utils/Markdown.java (hunk 2 above)
Comment (tdakanalis) [reply src/main/java/network/radicle/tools/github/utils/Markdown.java]: @JChrist Fixed. Could you review the changes please?

**Review by JChrist — APPROVED** [review]

**Pull request merged**